src/code/modules/remediation.js

```diff
@@ -417,12 +417,126 @@ function getEggSortRemediation(trait, /* practiceCriteria */) {
+    singleParent = false;
+  }
+  let baseAlleles = "a:W,b:W,a:M,b:M,a:T,b:T,a:H,b:H,a:Hl,b:Hl,a:Fl,b:Fl,a:A1,b:A1,a:C,b:C,a:Bog,b:Bog,a:Rh,b:Rh";
+  if (trait === "black") {
+    baseAlleles = baseAlleles + ",a:B,b:B,a:D,b:D";
+  } else {
+    baseAlleles = baseAlleles + ",a:b,b:b,a:d,b:d";
+  }
+  baseAlleles = baseAlleles.replace(GeneticsUtils.getAllelesForTrait(trait, "dominant") + ",", "");
+
+  let motherAlleles, fatherAlleles, child1Alleles, child2Alleles;
+  if (!isSiblings) {
+    if (trait === "armor" && practiceCriteria === "Incomplete Dominance") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, singleParent ? "heterozygous" : "dominant");
+    } else if (practiceCriteria === "X Linked") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+    } else if (trait === "tail") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, singleParent ? "dominant" : "recessive");
+      child1Alleles = "a:Tk,b:t";
+    } else {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, singleParent ? "heterozygous" : "dominant");
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+    }
+  } else {
+    if (trait === "armor" && practiceCriteria === "Incomplete Dominance") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, "heterozygous");
+      child2Alleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+    } else if (practiceCriteria === "X Linked") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+      child2Alleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+    } else if (trait === "tail") {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      fatherAlleles = GeneticsUtils.getAllelesForTrait(trait, "heterozygous");
+      child1Alleles = "a:Tk,b:t";
+      child2Alleles = "a:t,b:t";
+    } else {
+      motherAlleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+      fatherAlleles = motherAlleles;
+      child1Alleles = GeneticsUtils.getAllelesForTrait(trait, "dominant");
+      child2Alleles = GeneticsUtils.getAllelesForTrait(trait, "recessive");
+    }
+  }
```

Comment:
As your opening comment suggests, it's tempting to try to find a more general/declarative way to represent this, but it's not obvious to me what that looks like and this is easy to follow (and working) so better just to leave it alone.

Reply:
I could see shrinking this down into a table. The specifics of the `if` clauses would make it seem a bit arbitrary, though.